File: 01/01.go
package main

import (
	"fmt"
	"io/ioutil"
	"strconv"
	"strings"
)

const inputFile = "input.txt"

func main() {
	data, _ := ioutil.ReadFile(inputFile)

	lines := strings.Split(string(data), "\n")
	ns := []int{}
	for _, line := range lines {
		n, _ := strconv.Atoi(line)
		ns = append(ns, n)
	}
	for i, n1 := range ns {
		for _, n2 := range ns[i:] {
			if n1+n2 == 2020 {
				fmt.Printf("%d + %d = %d\n", n1, n2, n1+n2)
				fmt.Printf("%d * %d = %d\n", n1, n2, n1*n2)
			}
		}
	}
}

File: 01/01.hs
main :: IO ()
main = do
  inputs <- lines <$> readFile "input.txt"
  let ns = map read inputs :: [Int]
  let result1 = head [a * b | a <- ns, b <- ns, a + b == 2020]
  let result2 = head [a * b * c | a <- ns, b <- ns, c <- ns, a + b + c == 2020]
  putStrLn $ "Part 1: " ++ show result1
  putStrLn $ "Part 2: " ++ show result2
